tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -4,79 +4,226 @@
 
 import re
 
+from enum import Enum, auto
+
 import host_tools.network as net_tools  # pylint: disable=import-error
 
 
-def test_1vcpu(test_microvm_with_ssh, network_config):
-    """Test CPU feature emulation with 1 vCPU."""
-    test_microvm = test_microvm_with_ssh
-    test_microvm.spawn()
+class CpuVendor(Enum):
+    """CPU vendors enum."""
 
-    # Set up the microVM with 1 vCPUs, 256 MiB of RAM, no network ifaces, and
-    # a root file system with the rw permission. The network interfaces is
-    # added after we get a unique MAC and IP.
-    test_microvm.basic_config(vcpu_count=1)
+    AMD = auto()
+    INTEL = auto()
 
-    _tap, _, _ = test_microvm.ssh_network_config(network_config, '1')
-    test_microvm.start()
-    expected_cpu_topology = {
-        "CPU(s)": "1",
-        "On-line CPU(s) list": "0",
-        "Thread(s) per core": "1",
-        "Core(s) per socket": "1",
-        "Socket(s)": "1",
-        "NUMA node(s)": "1"
-    }
-    _check_cpu_topology(test_microvm, expected_cpu_topology)
 
+def _setup_microvm(test_microvm_with_ssh, network_config, vcpu_count,
```

Comment:
I can't see this helper fn being used anywhere. Is it?

Reply:
Nope. I forgot it there. Removed.